code/modules/mining/boulder_processing/_boulder_processing.dm

```diff
@@ -0,0 +1,344 @@
+			tripped = TRUE
+	if(!tripped)
+		remove_boulder(chosen_boulder)
+		say("Nothing to process!")
+		return FALSE //we shouldn't spend more time processing a boulder with contents we don't care about.
+	use_power(BASE_MACHINE_ACTIVE_CONSUMPTION)
+	check_for_boosts() //Calls the relevant behavior for boosting the machine's efficiency, if able.
+	var/is_artifact = (istype(chosen_boulder, /obj/item/boulder/artifact)) //We need to know if it's an artifact so we can carry it over to the new boulder.
+	var/obj/item/boulder/disposable_boulder = new (src)
+	disposable_boulder.custom_materials = processable_ores
+	silo_materials.mat_container.insert_item(disposable_boulder, refining_efficiency, context = src)
+	qdel(disposable_boulder)
+
+	refining_efficiency = initial(refining_efficiency) //Reset refining efficiency to 100% now that we've processed any relevant ores.
+	if(!length(chosen_boulder.custom_materials))
+		playsound(loc, 'sound/weapons/drill.ogg', 50, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+		if(is_artifact)
+			points_held = round(points_held + MINER_POINT_MULTIPLIER) /// Artifacts give bonus points!
+		chosen_boulder.break_apart()
+		update_boulder_count()
+		return TRUE //We've processed all the materials in the boulder, so we can just destroy it in break_apart.
+
+	chosen_boulder.restart_processing_cooldown() //So that we don't pick it back up!
+	chosen_boulder.durability = rand(chosen_boulder.boulder_size, chosen_boulder.boulder_size + BOULDER_SIZE_SMALL) //Reset durability to a random value between the boulder's size and a little more.
+	remove_boulder(chosen_boulder)
+	return TRUE
+
+/**
+ * Accepts a boulder into the machine. Used when a boulder is first placed into the machine.
+ * @param new_boulder The boulder to be accepted.
+ */
+/obj/machinery/bouldertech/proc/accept_boulder(obj/item/boulder/new_boulder)
+	if(isnull(new_boulder))
+		return FALSE
+	if(boulders_contained.len >= boulders_held_max) //Full already
+		return FALSE
+	if(!istype(new_boulder)) //Can't be processed
+		return FALSE
+	if(!new_boulder.custom_materials) //Shouldn't happen, but just in case.
+		qdel(new_boulder)
+		playsound(loc, 'sound/weapons/drill.ogg', 50, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+		return FALSE
+	new_boulder.forceMove(src)
+	boulders_contained += new_boulder
+	SSore_generation.available_boulders -= new_boulder
+	START_PROCESSING(SSmachines, src) //Starts processing if we aren't already.
+	return TRUE
+
+/**
+ * Ejects a boulder from the machine. Used when a boulder is finished processing, or when a boulder can't be processed.
+ * @param drop_turf The location to eject the boulder to. If null, it will eject to the machine's drop_location().
+ * @param specific_boulder The boulder to be ejected.
+ */
+/obj/machinery/bouldertech/proc/remove_boulder(obj/item/boulder/specific_boulder, turf/drop_turf = null)
+	if(!specific_boulder)
+		CRASH("remove_boulder() called with no boulder!")
+	if(isnull(specific_boulder))
```

Comment:
You gotta pick one `if(!specific_boulder)` & `if(isnull(specific_boulder))` do the same thing